src/ansys/fluent/core/search.py

```diff
@@ -281,10 +280,18 @@ def _write_api_tree_file(api_tree_data: dict, api_object_names: list):
                 if sys.version_info[0] < 3
                 else wn.synsets(name, lang="eng")
             )
-            synset_names = []
+            synset_names = set()
             for api_object_name_synset in api_object_name_synsets:
-                synset_names.append(api_object_name_synset.name().split(".")[0])
-            all_api_object_name_synsets[name] = synset_names
+                synset_names.add(api_object_name_synset.name().split(".")[0])
+            if synset_names:
+                unrelated_synsets = set()
+                synset_names.discard(name)
+                for synset_name in synset_names:
+                    if synset_name not in api_object_names:
+                        unrelated_synsets.add(synset_name)
+                all_api_object_name_synsets[name] = list(
+                    synset_names.difference(unrelated_synsets)
+                )
         api_tree_data["all_api_object_name_synsets"] = all_api_object_name_synsets
 
         api_tree_file = _get_api_tree_data_file()
```

Comment:
From both the names `api_tree_file` and `_get_api_tree_data_file()`, I get the impression that `api_tree_file` is going to be a file object, but it's a path object.

Reply:
It's path object.

Reply:
Ok. I see. We will add `_path` suffix at both places. Thank you.